### ISSUE TITLE
Add option to open new windows under the current one in the same stack

### ISSUE THREAD

**Issue by minego** [paperwm/PaperWM#195]
It would be really useful to have an option to open a new window under the current one, in the same stack, instead of opening it at full height to the right of the current window.

Right now if I want to open a new terminal and leave everything that is currently visible then I have to open a window, switch back to the previous window, and then hit ctrl+I to pull it into the stack.

With this option there would be cases where I want to immediately move a window out of a stack, but that is less effort than the other way around.

**Comment by olejorgenb** [comment]
I like this idea. It could be configurable per app type and number of windows already in the column("stack"). Eg. vertically tile ("stack") a new terminal if the current window also is a terminal and the column only have one window.

We should probably also have actions for moving/tiling the **current** window into the left/right column.

**Comment by minego** [comment]
I really like the idea of actions for moving the current window into the left or right column. That would feel much more natural to me.

**Comment by olejorgenb** [comment]
Pushed a quick custom keybinding: https://github.com/paperwm/PaperWM/commit/d2887263a27a421e00dbdb228f24bb149ba0cf9c

Pull master and apply this patch to your "~/.config/paperwm/user.js" (basically add a call to `Examples.keybindings.tileInto()`). Restart gnome-shell. <kbd>Super</kbd><kbd><</kbd> and <kbd>Super</kbd><kbd>></kbd> should tile the selected window into the left/right column. The bindings can be customized by passing two keystrings as arguments.

```diff
diff --git a/examples/user.js b/examples/user.js
index 7103f0e..e8f086e 100644
--- a/examples/user.js
+++ b/examples/user.js
@@ -21,6 +21,8 @@ function init() {
     Keybindings = Extension.imports.keybindings;
     Examples = Extension.imports.examples;
     App = Extension.imports.app;
+
+    Examples.keybindings.tileInto();
 }
 
 function enable() {
```

**Comment by minego** [comment]
Very nice, thank you!

It might be worth tweaking a bit more. Right now moving a window into a stack seems to select another window.

I think what I'd really like is a keybinding that will move a window left or right, going in and out of stacks as needed. So, if the window is not in a stack and you try to move it right then it will move into the stack with the window(s) to the right of it, and stay active. If you then try to move it right again that will pull it out of that stack and into it's own.

**Comment by olejorgenb** [comment]
Hm, that what `tileInto` is supposed to do. The window should stay active. I'll look into it when I get home. 

> If you then try to move it right again that will pull it out of that stack and into it's own

Good idea, that could be a good compromise to avoid having a dedicated untile-current-left/right command

**Comment by minego** [comment]
It is possible that the window selection is being weird because I have focus follow mouse enabled. From what I've seen so far there are a few cases where PaperWM doesn't entirely play nicely with that. I'd really like an option to warp the mouse to a window when it is selected or moved to prevent accidentally selecting windows that I don't mean to. Either that, or somehow disable mouse focus when the mouse hasn't actually moved. I'm not sure if those things really belong as part of PaperWM though.

**Comment by olejorgenb** [comment]
Yup, it's the "sloppy focus". Pointer warping could probably be implemented as an separate extension (note that pointer warping doesn't yet work on wayland)

**Comment by minego** [comment]
> Yup, it's the "sloppy focus". Pointer warping could probably be implemented as an separate extension (note that pointer warping doesn't yet work on wayland)

Good to know. I haven't been able to find an extension for pointer warping, so I may attempt to write my own when I get a chance. I'm not running wayland (yet) anywhere but that is an important thing to know.

Perhaps the better approach would be a smarter mouse focus mode. It should not change the selected window if the mouse didn't actually move.

**Comment by Thesola10** [comment]
This has been addressed by the introduction of #862 

---

_🧹 Just going through and triaging old issues..._
